lib/puppet/resource.rb

```diff
@@ -255,7 +255,7 @@ def initialize(type, title = nil, attributes = EMPTY_HASH)
           "Did you mean (#{(type[:type] || type["type"]).inspect}, #{(type[:title] || type["title"]).inspect }) ?"
       end
 
-      environment = attributes[:environment]
+      self.environment = attributes[:environment]
```

Comment:
@hlindberg @thallgren This seems particularly bad.

Reply:
Looks like that line has 0 effect. There is however a block of code further down:
```
      # Set things like strictness first.
      attributes.each do |attr, value|
        next if attr == :parameters
        send(attr.to_s + "=", value)
      end
```
and I think there is an `environment=(x)` method that gets triggered if there is an 'environment' key in the attributes hash.